### PR TITLE
Add KafkaContainer#withListener

### PR DESCRIPTION
**Pull request by eddumelendez**
Allow to register additional listeners. E.g. when using along with
Toxiproxy, Schema Registry, Kafka Connect, KCat.

Listeners's host will register as a network aliases.

### CODE REVIEW

File: docs/modules/kafka.md [hunk 1]
```diff
@@ -42,18 +42,27 @@ KRaft mode was declared production ready in 3.3.1 (confluentinc/cp-kafka:7.3.x)"
 
 See the [versions interoperability matrix](https://docs.confluent.io/platform/current/installation/versions-interoperability.html) for more details. 
 
-## Multi-container usage
+## Register listeners
 
-If your test needs to run some other Docker container which needs access to Kafka, do the following:
+There are scenarios where connecting to the broker is needed. The consumer/producer can be another container
+in the same network or a different process where port to connect to is different from the exposed port `9093`.
```
Comment (kiview) [inline docs/modules/kafka.md:48]:
```suggestion
in the same network or a different process where the port to connect differs from the default exposed port `9093`.
```

File: docs/modules/kafka.md [hunk 2]
```diff
@@ -42,18 +42,27 @@ KRaft mode was declared production ready in 3.3.1 (confluentinc/cp-kafka:7.3.x)"
...
-Or just give your Kafka container a network alias of your liking.
 
-You will need to explicitly create a network and set it on the Kafka container as well as on your other containers that need to communicate with Kafka.
+Container defined in the same network:
+
+<!--codeinclude-->
+[Create kcat container](../../modules/kafka/src/test/java/org/testcontainers/containers/KafkaContainerTest.java) inside_block:createKCatContainer
+<!--/codeinclude-->
+
+Client using the new listener registered:
```
Comment (kiview) [inline docs/modules/kafka.md:61]:
```suggestion
Client using the new registered listener:
```

File: modules/kafka/src/main/java/org/testcontainers/containers/KafkaContainer.java [hunk 3]
```diff
@@ -230,6 +276,31 @@ protected String commandZookeeper() {
         return command;
     }
 
+    /**
+     * Add a {@link Supplier} that will provide a listener with format {@code host:port}.
+     * Host will be added as a network alias.
+     * <p>
+     * The listener will be added to the existing listeners.
```
Comment (kiview) [inline modules/kafka/src/main/java/org/testcontainers/containers/KafkaContainer.java:283]:
```suggestion
     * The listener will be added to the list of default listeners.
```

File: modules/kafka/src/main/java/org/testcontainers/containers/KafkaContainer.java [hunk 4]
```diff
@@ -230,6 +276,31 @@ protected String commandZookeeper() {
...
+     * <p>
+     * Existing listeners:
+     * <ul>
+     *     <li>0.0.0.0:9092</li>
+     *     <li>0.0.0.0:9093</li>
+     * </ul>
+     * <p>
+     * Existing advertised listeners:
```
Comment (kiview) [inline modules/kafka/src/main/java/org/testcontainers/containers/KafkaContainer.java:291]:
```suggestion
     * Default advertised listeners:
```

File: docs/modules/kafka.md [hunk 5]
```diff
@@ -42,18 +42,27 @@ KRaft mode was declared production ready in 3.3.1 (confluentinc/cp-kafka:7.3.x)"
...
 
-If your test needs to run some other Docker container which needs access to Kafka, do the following:
+There are scenarios where connecting to the broker is needed. The consumer/producer can be another container
+in the same network or a different process where port to connect to is different from the exposed port `9093`.
+E.g [Toxiproxy](../../docs/modules/toxiproxy.md).
 
-* Run your other container on the same network as Kafka container, e.g.:
 <!--codeinclude-->
-[Network](../../modules/kafka/src/test/java/org/testcontainers/containers/KafkaContainerTest.java) inside_block:withKafkaNetwork
+[Register listener](../../modules/kafka/src/test/java/org/testcontainers/containers/KafkaContainerTest.java) inside_block:registerListener
```
Comment (kiview) [inline docs/modules/kafka.md:52]:
```suggestion
[Register additional listener](../../modules/kafka/src/test/java/org/testcontainers/containers/KafkaContainerTest.java) inside_block:registerListener
```

**Review by kiview — DISMISSED** [review]

File: docs/modules/kafka.md [hunk 6]
```diff
@@ -42,18 +42,27 @@ KRaft mode was declared production ready in 3.3.1 (confluentinc/cp-kafka:7.3.x)"
 
 See the [versions interoperability matrix](https://docs.confluent.io/platform/current/installation/versions-interoperability.html) for more details. 
 
-## Multi-container usage
+## Register listeners
 
-If your test needs to run some other Docker container which needs access to Kafka, do the following:
+There are scenarios where connecting to the broker is needed. The consumer/producer can be another container
```
[the remark this answers is not in the document]
Comment (eddumelendez) [reply docs/modules/kafka.md]: I've updated this section

**Pull request merged**